### ISSUE TITLE
Double BG-dots on site 

### ISSUE THREAD

**Issue by whooze** [pazaan/600SeriesAndroidUploader#124]
Ran the new uploader tonight and it's been working like a charm, didn't miss a single reading!
I did a calibration and gave a combo-bolus before I went to bed, it showed up as one red dot on my nightscout-site.

This morning, before leaving for work I gave a combo-bolus and it shows on the site as two dots, 5 minutes apart.

Using a Samsung S6 as uploader, connected with WiFi.
Don't know about the android version and stuff now since I'm at work.

![image](https://user-images.githubusercontent.com/4024639/27900074-156b3ee6-622d-11e7-8683-b61f00ecf9df.png)

**Comment by volkerrichert** [comment]
hm, I added some code to avoid this. We'll check it

**Comment by pazaan** [comment]
@volkerrichert - that might have gotten a bit messed up with all of the cross-merging before the release - sorry!

**Comment by volkerrichert** [comment]
yeap, that was also my idea.

**Comment by volkerrichert** [comment]
@whooze one question: How do you enable the dots in the chart above? I only have them in the lower chart.

**Comment by whooze** [comment]
Ummm, as far as I remember it has always been like that for me
![image](https://user-images.githubusercontent.com/4024639/27900559-257b7632-622f-11e7-8306-e75f647c0ea3.png)

Maybe I'm just dreaming but I belive I could see active insulin automaticly with the earlier version of the uploader. Without having to add the dose manually to Nightscout.

**Comment by pazaan** [comment]
I have the following modules in my `ENABLE`: `careportal cage basal iob cob bwp treatmentnotify pump loop bridge`
I'm not sure exactly which one shows the SMBGs.

**Comment by pazaan** [comment]
Maybe it's `treatmentnotify`?

**Comment by whooze** [comment]
I had
`careportal rawbg cage sage pump cob boluscalc pushover iob maker`

Now I have
`careportal cage sage pump cob boluscalc pushover iob maker treatmentnotify`
(no need to have rawbg these days)

Gives me this
![image](https://user-images.githubusercontent.com/4024639/27901125-2897af6e-6231-11e7-8da6-ac7bf07dbde1.png)

![image](https://user-images.githubusercontent.com/4024639/27901145-38229cdc-6231-11e7-9e2b-756fe6d646b2.png)

It should be some insulin active I belive. Will troubleshoot this more tonight and maybe a bit during the day when she's getting up for breakfast

**Comment by volkerrichert** [comment]
treatmentnotify does not work. i added all of your modules and reset the settings

![image](https://user-images.githubusercontent.com/640914/27901904-105ec632-6234-11e7-9793-e78abcc68ac8.png)

Strange, very ...

**Comment by volkerrichert** [comment]
but that's not the point. As you see. I just have a single dot.

**Comment by pazaan** [comment]
I have seen double-dots as well. Since I'm running in an emulator attached to a debugger, I'll see if I can figure out what causes them.

**Comment by volkerrichert** [comment]
yeap, i have seem them also.

**Comment by volkerrichert** [comment]
you already told me the cause week ago. The value will appear in the communication for 15 min. i added code to check if it increase.

**Comment by pazaan** [comment]
@whooze - do you recall how much insulin was given "up front" with the combo bolus? Is it possible that no insulin was delivered, and that everything else was in the square wave?

**Comment by whooze** [comment]
@pazaan Since I'm not cable of going to sleep when I should I'm all dizzy in the morning but yeah, I actually think that I set 0% now and 100% in 30 minutes or something like that. 
Called the daughter and asked and she can't see it in the pump :(

I can upload the pump to carelink tonight and confirm

**Comment by whooze** [comment]
I noticed she had left the bedroom cause there was a dot missing on nightscout so I called her.

She was in the living-room, so no connection to the uploader.

The uploader kinda refused to read her pump again. Waited until there was 15 minutes until last reading then asked her to manually poll for new data and that worked. Don't want to trouble her with logs from the phone and stuff.

Will try to reproduce it tonight. 

We gave a combo-bolus with 0% now and 100% in 30 minutes. This time it shows up as one dot, but that could ofcourse be because of the lost connection.

Got the IOB now though!
![image](https://user-images.githubusercontent.com/4024639/27905024-f392043c-623e-11e7-98e5-f2ff0411d46f.png)
